File: examples/python/runner/runner_mcp.py
"""Example: MCP-backed runner that hosts filesystem and context7 MCP servers."""

import os

from rebuno import MCPServer, Runner

if __name__ == "__main__":
    runner = Runner(
        os.getenv("RUNNER_ID", "mcp-tools"),
        kernel_url=os.getenv("REBUNO_KERNEL_URL", "http://localhost:8080"),
    )

    # Stdio MCP server
    runner.host(MCPServer(
        "filesystem",
        command="npx",
        args=["-y", "@modelcontextprotocol/server-filesystem", "/tmp"],
    ))

    # HTTP MCP server
    runner.host(MCPServer("context7", url="https://mcp.context7.com/mcp"))

    runner.run()

File: examples/typescript/runner/runner_mcp.ts
// Example: MCP-backed runner that wraps filesystem and fetch MCP servers.

import { BaseRunner } from "rebuno";

const runner = new (class extends BaseRunner {})({
  runnerId: process.env.RUNNER_ID ?? "mcp-tools",
  kernelUrl: process.env.REBUNO_KERNEL_URL ?? "http://localhost:8080",
});

// Stdio MCP server
runner.mcpServer({
  name: "filesystem",
  command: "npx",
  args: ["-y", "@modelcontextprotocol/server-filesystem", "/tmp"],
});

// HTTP MCP server
runner.mcpServer({
  name: "context7",
  url: "https://mcp.context7.com/mcp",
});

runner.run();
